csharp/src/Ice/Reference.cs

```diff
@@ -608,18 +624,31 @@ public override string ToString()
                     sb.Append(TimeSpanExtensions.ToPropertyValue(invocationTimeout));
                 }
 
+                if (Label != null)
+                {
+                    StartQueryOption(sb, ref firstOption);
+                    sb.Append("label=");
+                    sb.Append(Label.ToString());
+                }
+
                 if (_locatorCacheTimeout is TimeSpan locatorCacheTimeout)
                 {
                     StartQueryOption(sb, ref firstOption);
                     sb.Append("locator-cache-timeout=");
                     sb.Append(TimeSpanExtensions.ToPropertyValue(locatorCacheTimeout));
                 }
 
-                if (_preferNonSecure is bool preferNonSecure)
+                if (!PreferExistingConnection)
+                {
+                    StartQueryOption(sb, ref firstOption);
+                    sb.Append("prefer-existing-connection=false");
+                }
```

Comment:
Shouldn't this be handled like other proxy options and be printed only if different from `Communicator.DefaultPreferExistingConnection`?

Reply:
We do the same for `CacheConnection` which is the most similar to this

https://github.com/zeroc-ice/ice/blob/f1ed3c9e2b1a26c0a9ab766921b48a086d338e38/csharp/src/Ice/Reference.cs#L584

Reply:
I did it that way because there is no `Ice.Default.CacheConnection`: the default is always true and cannot be changed.
So if we add a `Ice.Default.PerferExistingConnection`, we should output this value when it's different than `Communicator.DefaultPerferExistingConnection`. And use a nullable _preferExistingConnection field.